java-frontend/src/main/java/org/sonar/java/AbstractJavaClasspath.java

```diff
@@ -102,6 +103,10 @@ protected boolean hasJavaSources() {
     return fs.hasFiles(fs.predicates().and(fs.predicates().hasLanguage("java"), fs.predicates().hasType(fileType)));
   }
 
+  protected boolean hasMoreThanOneJavaFile() {
+    return Iterables.size(fs.files(fs.predicates().and(fs.predicates().hasLanguage("java"), fs.predicates().hasType(fileType)))) > 1;
```

Comment:
Okay, so let's continue to post warning in case of missing byte code for test sources, but start failing on main sources.